src/tr2/decomp/skidoo.c

```diff
@@ -874,20 +874,16 @@ int32_t Skidoo_Control(void)
 
     if (skidoo->flags & IF_ONE_SHOT) {
         Room_TestTriggers(g_LaraItem);
-        if (room_num != skidoo->room_num) {
-            Item_NewRoom(g_Lara.skidoo, room_num);
-        }
+        Item_UpdateRoom(g_Lara.skidoo, room_num);
         if (skidoo->pos.y == skidoo->floor) {
             Skidoo_Explode(skidoo);
         }
         return 0;
     }
 
     Skidoo_Animation(skidoo, collide, dead);
-    if (room_num != skidoo->room_num) {
-        Item_NewRoom(g_Lara.skidoo, room_num);
-        Item_NewRoom(g_Lara.item_num, room_num);
-    }
+    Item_UpdateRoom(g_Lara.skidoo, room_num);
+    Item_UpdateRoom(g_Lara.item_num, room_num);
```

Comment:
Flagging it as a risky update too.

Reply:
Lara and the skidoo always have the same room number - I don't feel this is risky, unless I'm perhaps missing something?

Reply:
I haven't fully followed the code, only saw there was no `if` and wanted us to pay extra attention to it if `room_num` of both items can be relied on in this case.